File: DevExpress-Demo/VB/LayoutControlDemo.Wpf/ViewModels/RealEstateViewModel.vb
Imports System.Collections.Generic
Imports System.Linq

Namespace LayoutControlDemo

    Public Class RealEstateViewModel

        Public Sub New()
            Source = Listings.DataSource
            SelectedListing = Source.First()
        End Sub

        Private Property Source As IList(Of Listing)

        Public Overridable Property SelectedListing As Listing

        Public Sub NextListing()
            Dim index = SelectedIndex()
            SelectedListing = Source(System.Threading.Interlocked.Increment(index))
        End Sub

        Public Function CanNextListing() As Boolean
            Return SelectedListing IsNot Source.Last()
        End Function

        Public Sub PreviousListing()
            Dim index = SelectedIndex()
            SelectedListing = Source(System.Threading.Interlocked.Decrement(index))
        End Sub

        Public Function CanPreviousListing() As Boolean
            Return SelectedListing IsNot Source.First()
        End Function

        Private Function SelectedIndex() As Integer
            Return Source.IndexOf(SelectedListing)
        End Function
    End Class
End Namespace

File: DevExpress-Demo/CS/LayoutControlDemo.Wpf/ViewModels/RealEstateViewModel.cs
using System.Collections.Generic;
using System.Linq;

namespace LayoutControlDemo {
    public class RealEstateViewModel {
        public RealEstateViewModel() {
            Source = Listings.DataSource;
            SelectedListing = Source.First();
        }
        IList<Listing> Source { get; set; }
        public virtual Listing SelectedListing { get; set; }

        public void NextListing() {
            var index = SelectedIndex();
            SelectedListing = Source[++index];
        }
        public bool CanNextListing() {
            return SelectedListing != Source.Last();
        }
        public void PreviousListing() {
            var index = SelectedIndex();
            SelectedListing = Source[--index];
        }
        public bool CanPreviousListing() {
            return SelectedListing != Source.First();
        }
        int SelectedIndex() {
            return Source.IndexOf(SelectedListing);
        }
    }
}
